client/components/theme-collection/index.tsx

```diff
@@ -0,0 +1,131 @@
+	isInstalling: () => boolean;
+}
+
+export default function ThemeCollection( {
+	collectionSlug,
+	heading,
+	subheading,
+	themes,
+	getScreenshotUrl,
+	siteId,
+	getButtonOptions,
+	getActionLabel,
+	isActive,
+	getPrice,
+	isInstalling,
+}: ThemeCollectionProps ): ReactElement {
+	const swiperInstance = useRef< SwiperType | null >( null );
+	useEffect( () => {
+		if ( themes !== null && themes.length > 0 ) {
+			const el = document.querySelector( '.swiper-container' ) as HTMLElement;
+			if ( el ) {
+				/**
+				 * We have to import the swiper package dynamically because it doesn't offer a CommonJS version. Because of this, the SSR build will fail.
+				 *
+				 * From a performance standpoint, I don't think it will cause any issues; we are already loading the ThemeCards dynamically anyway.
+				 */
+				Promise.all( [
+					import( 'swiper' ),
+					import( 'swiper/modules' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/navigation' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/keyboard' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/mousewheel' ),
+				] ).then( ( values ) => {
+					const Swiper = values[ 0 ].default;
+					const { Navigation, Keyboard, Mousewheel } = values[ 1 ];
+
+					swiperInstance.current = new Swiper( el, {
+						cssMode: true,
+						mousewheel: true,
+						keyboard: {
+							enabled: true,
+							onlyInViewport: false,
+						},
+						navigation: {
+							nextEl: '.theme-collection__carousel-nav-button--next',
+							prevEl: '.theme-collection__carousel-nav-button--previous',
+						},
+						threshold: 5,
+						slideToClickedSlide: true,
+						slidesPerView: 'auto',
+						spaceBetween: 20,
+						modules: [ Navigation, Keyboard, Mousewheel ],
+					} );
+				} );
+			}
+		}
+
+		return () => {
+			swiperInstance.current?.destroy();
+		};
+	}, [ themes ] );
+
+	if ( themes === null ) {
+		return <></>; // @todo Add LoadingPlaceholders instead.
+	}
+
+	return (
+		<div className="theme-collection__container ">
+			<h2>{ heading }</h2>
+			{ subheading }
+			<div className="swiper-container">
+				<div className="theme-collection__carousel-controls">
+					<Button className="theme-collection__carousel-nav-button theme-collection__carousel-nav-button--previous">
+						<Icon icon={ chevronLeft } />
+					</Button>
+					<Button className="theme-collection__carousel-nav-button theme-collection__carousel-nav-button--next">
+						<Icon icon={ chevronRight } />
+					</Button>
+				</div>
+				<div className="theme-collection__list-wrapper swiper-wrapper">
+					{ themes.map( ( theme, index ) => (
+						<div
+							key={ `theme-collection-container-${ collectionSlug }-${ index }` }
```

Comment:
`index` is just the array index, react keys should use constant indexes not numeric array indexes because the zeroth item on the first render, could be the first item on the second render and it would update the wrong element.

This is probably only going to be a problem in this case if someone nulls out the react state and then someone changes curation order and so it refetches things, or something.

But still, easy enough to use theme.slug or something to generate a key.

Reply:
Sure, I think we can change it. The numeric approach was taken from the existing implementation of the theme list component.